docs/deployment/uwsgi_cookiecutter_nginx.rst

```diff
@@ -0,0 +1,233 @@
+        from pyramid.scripts.common import get_config_loader
+        app_name    = 'main'
+        config_vars = {}
+        config_uri  = 'production.ini'
+
+        loader = get_config_loader(config_uri)
+        loader.setup_logging(config_vars)
+        app = loader.get_wsgi_app(app_name, config_vars)
+
+    `config_uri` is the project configuration file name.  It's best to use
+    the ``production.ini`` file provided by your cookiecutter, as it contains
+    settings appropriate for production.  `app_name` is the name of the section
+    within the ``.ini`` file that should be loaded by ``uWSGI``.  The
+    assignment to the name ``app`` is important: we will reference ``app`` and
+    the name of the file, ``wsgi`` when we invoke uWSGI.
+
+    The call to :func:`loader.setup_logging` initializes the standard
+    library's `logging` module to allow logging within your application.
+    See :ref:`logging_config`.
+
+#.  Create a new directory at ``~/myproject/tmp`` to house a pidfile and a unix
+    socket.  However, you'll need to make sure that *two* users have access to
+    change into the ``~/myproject/tmp`` directory: your current user (mine is
+    ``ubuntu`` and the user that Nginx will run as often named ``www-data`` or
+    ``nginx``).
+
+#.  Invoke uWSGI.
+
+    .. code-block:: bash
+
+      cd ~/myproject
+      sudo uwsgi \
+        --chmod-socket=020 \
+        --enable-threads \
+        --plugin=python3 \
+        --socket ~/myproject/tmp/myproject.sock \
+        --manage-script-name \
+        --mount /=wsgi:app \
+        --uid ubuntu \
+        --gid www-data \
+        --virtualenv env
+
+      # Explanation of Options
+      # sudo uwsgi                          # Invoke as sudo so you can masquerade
+      #                                       as the users specfied in --uid and --gid
+      #
+      # --chmod-socket=020                  # Change permissions on socket to
+      #                                       at least 020 so that in combination
+      #                                       with "--gid www-data", Nginx will be able
+      #                                       to write to it after  uWSGI creates it
+      #
+      # --enable-threads                    # Execute threads that are in your app
+      #
+      # --plugin=python3                    # Use the python3 plugin
+      #
+      # --socket ~/myproject/tmp/myproject.sock   # Where to put the unix socket
+      #
+      # --manage-script-name
+      #
+      # --mount /=wsgi:app                  # Mount the path "/" on the symbol
+      #                                       "app" found in the file wsgi.py
+      #
+      # --uid ubuntu                        # masquerade as the ubuntu user
+      #
+      # --gid www-data                      # masquerade as the www-data group
+      #
+      # --virtualenv env                    # Use packages installed in your venv
```

Comment:
this options can be easily added in .ini file of pyramid app under `[uwsgi]` section. I think it makes it more clean to put it there.